src/reference/asciidoc/gateway.adoc

```diff
@@ -764,6 +764,30 @@ mono.subscribe(invoice -> handleInvoice(invoice));
 
 The calling thread continues, with `handleInvoice()` being called when the flow completes.
 
+===== Downstream Flows Returning an Async Type
+
+As mentioned in the `ListenableFuture` section above; if you wish some downstream component to return a message with an async payload (`Future`, `Mono`, etc), you must explicitly set the async executor to `null` (or `""` when using XML configuration).
+The flow is then invoked on the caller thread and the result can be retrieved later.
+
+===== void Return Type
```

Comment:
Or `void` (I mean wrapped into the code) or a "A Void Return Type, if it is about supply the style @Buzzardo suggests to us...